zerver/lib/bugdown/__init__.py

```diff
@@ -1101,6 +1111,7 @@ def extendMarkdown(self, md, md_globals):
                         \*\*                         # ends by double asterisks
                        """
         md.inlinePatterns.add('stream', StreamPattern(stream_group), '>backtick')
+        md.inlinePatterns.add('katex', KatexPattern(r'(?P<delim>\${1,2})(?P<body>[^$]+)(?P=delim)'), '>backtick')
```

Comment:
Yeah, the idea is to leave that more clear and properly organised (with some explanations on what does the regex do.)

However, two of the tasks in the to-do list require modifying that line, so I'm going to leave it like that until we have everything sorted out. Thanks! :)